File: 3sem/4lab/Banks.Console/Handlers/ShowBanksHandler.cs
using Banks.Entities;
using Banks.Services;

namespace Banks.Console.Handlers;

public class ShowBanksHandler : Handler
{
    public override void Handle(string request, CentralBank centralBank)
    {
        ArgumentNullException.ThrowIfNull(request);
        ArgumentNullException.ThrowIfNull(centralBank);

        if (request.ToLower() == "show banks")
        {
            foreach (Bank bank in centralBank.Banks)
                System.Console.WriteLine($" * Bank '{bank.Name}' {bank.Id}");

            return;
        }

        NextHandler?.Handle(request, centralBank);
    }
}

File: 4sem/1lab/src/main/java/BanksRemake/Presentation/Handlers/ShowBanksHandler.java
package banksRemake.presentation.handlers;

import banksRemake.domain.entities.Bank;
import banksRemake.domain.exceptions.InvalidBankAccountOperationException;
import banksRemake.domain.exceptions.InvalidBankOperationException;
import banksRemake.domain.exceptions.InvalidDepositInterestRateSelectorException;
import banksRemake.domain.exceptions.InvalidMoneyGapException;
import banksRemake.domain.services.CentralBank;

import java.util.Scanner;

public class ShowBanksHandler extends HandlerBase {
    @Override
    public void handle(String request, CentralBank centralBank, Scanner scanner) throws InvalidBankOperationException, InvalidBankAccountOperationException, InvalidDepositInterestRateSelectorException, InvalidMoneyGapException {
        if (request.toLowerCase().equals("show banks")) {
            for (Bank bank : centralBank.getBanks()) {
                System.out.println(String.format(" * Bank '%s' %s", bank.getName(), bank.getId()));
            }

            return;
        }

        if (getNext() != null) {
            getNext().handle(request, centralBank, scanner);
        }
    }
}
